src/ProjectOrigin.WalletSystem.IntegrationTests/Repositories/CertificateRepositoryTests.cs

```diff
@@ -174,6 +174,85 @@ public async Task GetAllOwnedCertificates()
+        var certificate1 = await CreateCertificate(registry);
+        var certificate2 = await CreateCertificate(registry, GranularCertificateType.Consumption);
+        var certificate3 = await CreateCertificate(registry);
+        var owner1 = _fixture.Create<string>();
+        var wallet1 = await CreateWallet(owner1);
+        var endpoint1 = await CreateWalletEndpoint(wallet1);
+        var endpoint2 = await CreateWalletEndpoint(wallet1);
+        var owner2 = _fixture.Create<string>();
+        var wallet2 = await CreateWallet(owner2);
+        var endpoint3 = await CreateWalletEndpoint(wallet2);
+        //Wallet1
+        var slice1 = new WalletSlice
+        {
+            Id = Guid.NewGuid(),
+            WalletEndpointId = endpoint1.Id,
+            WalletEndpointPosition = endpointPosition,
+            RegistryName = registry,
+            CertificateId = certificate1.Id,
+            Quantity = _fixture.Create<int>(),
+            RandomR = _fixture.Create<byte[]>(),
+            State = WalletSliceState.Available
+        };
+        var slice2 = new WalletSlice
+        {
+            Id = Guid.NewGuid(),
+            WalletEndpointId = endpoint1.Id,
+            WalletEndpointPosition = endpointPosition + 1,
+            RegistryName = registry,
+            CertificateId = certificate1.Id,
+            Quantity = _fixture.Create<int>(),
+            RandomR = _fixture.Create<byte[]>(),
+            State = WalletSliceState.Available
+        };
+        //Certficiate2
+        var slice3 = new WalletSlice
+        {
+            Id = Guid.NewGuid(),
+            WalletEndpointId = endpoint2.Id,
+            WalletEndpointPosition = endpointPosition,
+            RegistryName = registry,
+            CertificateId = certificate2.Id,
+            Quantity = _fixture.Create<int>(),
+            RandomR = _fixture.Create<byte[]>(),
+            State = WalletSliceState.Available
+        };
+
+        var sliceWithDifferentOwner = new WalletSlice
+        {
+            Id = Guid.NewGuid(),
+            WalletEndpointId = endpoint3.Id,
+            WalletEndpointPosition = endpointPosition,
+            RegistryName = registry,
+            CertificateId = certificate3.Id,
+            Quantity = _fixture.Create<int>(),
+            RandomR = _fixture.Create<byte[]>(),
+            State = WalletSliceState.Available
+        };
+
+        await _repository.InsertWalletSlice(slice1);
+        await _repository.InsertWalletSlice(slice2);
+        await _repository.InsertWalletSlice(slice3);
+        await _repository.InsertWalletSlice(sliceWithDifferentOwner);
+
+        var certificates = await _repository.GetAllOwnedCertificates(owner1, new CertificatesFilter
+        {
+            Type = GranularCertificateType.Consumption
+        });
+
+        certificates.Should().HaveCount(1).And.Satisfy(
+            c => c.Id == certificate2.Id && c.Slices.Sum(x => x.Quantity) == slice3.Quantity
```

Comment:
Naming the slices without a clear indication of the certificate it is a slice of, makes this a bit difficult to understand.